src/compiler/utilities.ts

```diff
@@ -3686,7 +3686,8 @@ namespace ts {
      * @param pos The start position.
      * @param end The end position.
      */
-    export function createRange(pos: number, end: number): TextRange {
+    export function createRange(pos: number, end: number = pos, noAssert = false): TextRange {
+        Debug.assert(noAssert || end >= pos);

```

Comment:
We treat `-1` as a special value to indicate a synthetic position. Rather than adding a `noAssert`, we could just write:

```ts
Debug.assert(pos === -1 || end === -1 || end >= pos);
```